### PR TITLE
feat: add merge_strategy support for coder_env resources

### PR DESCRIPTION
**Pull request by kacpersaw**
## Description

Implements the server-side merge logic for the `merge_strategy` attribute added to `coder_env` in [terraform-provider-coder v2.15.0](https://github.com/coder/terraform-provider-coder/pull/489). This allows template authors to control how duplicate environment variable names are combined across multiple `coder_env` resources.

Relates to https://github.com/coder/coder/issues/21885

## Supported strategies

| Strategy | Behavior |
|----------|----------|
| `replace` (default) | Last value wins — backward compatible |
| `append` | Joins values with `:` separator (e.g. PATH additions) |
| `prepend` | Prepends value with `:` separator |
| `error` | Fails the build if the variable is already defined |

## Example

```hcl
resource "coder_env" "path_tools" {
  agent_id       = coder_agent.dev.id
  name           = "PATH"
  value          = "/home/coder/tools/bin"
  merge_strategy = "append"
}
```

## Changes

- **Proto**: Added `merge_strategy` field to `Env` message in `provisioner.proto`
- **State reader**: Updated `agentEnvAttributes` struct and proto construction in `resources.go`
- **Merge logic**: Added `mergeExtraEnvs()` function in `provisionerdserver.go` with strategy-aware merging for both agent envs and devcontainer subagent envs
- **Tests**: 15 unit tests covering all strategies, edge cases (empty values, mixed strategies, multiple appends)
- **Dependency**: Bumped `terraform-provider-coder` v2.14.0 → v2.15.0
- **Fixtures**: Updated `duplicate-env-keys` test fixtures and golden files

## Ordering

When multiple resources `append` or `prepend` to the same key, they are processed in alphabetical order by Terraform resource address (per the determinism fix in #22706).

### CODE REVIEW

**Review by johnstcn — COMMENTED** [review]
Need to bump `CurrentMinor`.

File: docs/admin/templates/extending-templates/environment-variables.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+# This value is ignored because coder_agent.dev.env sets PATH directly.
+resource "coder_env" "extra_path" {
+  agent_id       = coder_agent.dev.id
+  name           = "PATH"
+  value          = "/home/coder/bin"
+  merge_strategy = "append"
+}
+```
```
Comment (johnstcn) [inline docs/admin/templates/extending-templates/environment-variables.md:115]: So this doesn't work with `merge_strategy = "error"`?

**Review by johnstcn — APPROVED** [review]

File: docs/admin/templates/extending-templates/environment-variables.md (hunk 1 above)
Comment (kacpersaw) [reply docs/admin/templates/extending-templates/environment-variables.md:115]: Updated docs @johnstcn
